torchgeometry/conversions.py

```diff
@@ -115,8 +115,16 @@ def convert_points_to_homogeneous(points):
         raise ValueError("Input must be at least a 2D tensor. Got {}".format(
             points.shape))
 
+    if torch.is_tensor(points.shape[-1]):
+        last_dim_shape = points.shape[-1].float()
+        last_dim_bit_length = torch.floor(torch.log2(last_dim_shape)) + 1
+        last_dim_bit_length = last_dim_bit_length.long()
```

Comment:
gotcha. However, I think we are over complicating things, similar to your other proposal with just padding the tensor it should work right ? `torch.nn.functional.pad(x, (0, 1), "constant", 1.0)`

Reply:
We can definitely pad the tensor, instead of concatenating it with ones. However, we still need to calculate bit length of the last dimension in order to determine the size of the padding.

Reply:
even another option to avoid bit length compute would be something like that:
`return torch.cat([points, torch.ones_like(points)[..., :1]], dim=-1)`

we are allocating unnecessary memory but I don't think I will harm the performance at all and it's clean.

Reply:
or wait, that one might work better: D
`return torch.cat([points, torch.ones_like(points)[..., -1:]], dim=-1)`

Reply:
anyways, whatever approach you choose please provide a unittest for JIT

Reply:
I have made a commit that removes bit length computation and constructs the tensor via padding. I will try to provide JIT versions of the unit tests.